diffkemp/building/build_kernel.py

```diff
@@ -80,46 +93,58 @@ def generate_from_sysctl_list(snapshot, sysctl_list):
         for sysctl in sysctl_list:
             print("{}:".format(sysctl))
 
-            # Proc handler function for sysctl
             proc_fun = sysctl_mod.get_proc_fun(sysctl)
-            if proc_fun:
-                try:
-                    proc_fun_mod = snapshot.source_tree.get_module_for_symbol(
-                        proc_fun)
-                    snapshot.add_fun(name=proc_fun,
-                                     llvm_mod=proc_fun_mod,
-                                     glob_var=None,
-                                     tag="proc handler function",
-                                     group=sysctl)
-                    print("  {}: {} (proc handler)".format(
-                        proc_fun,
-                        os.path.relpath(proc_fun_mod.llvm,
-                                        snapshot.source_tree.source_dir)))
-                except SourceNotFoundException:
-                    print("  could not build proc handler")
-
+            # Proc handler function for sysctl
+            _add_proc_handler(snapshot, sysctl, proc_fun)
             # Functions using the sysctl data variable
-            data = sysctl_mod.get_data(sysctl)
-            if not data:
+            _add_data_func(snapshot, sysctl, sysctl_mod, proc_fun)
```

Comment:
This method could use a better name, at least `_add_data_funcs` to highlight that multiple functions are potentially added. But perhaps even better is something like `_add_funcs_for_data` since it is adding functions which use the data variable.